pkg/sql/logictest/testdata/logic_test/temp_table

```diff
@@ -227,3 +227,26 @@ query I
 SELECT * FROM regression_47030
 ----
 2
+
+# renaming a temp table should not move it to the public schema.
+subtest regression_48233
+
+statement ok
+CREATE TEMP TABLE regression_48233(a int)
+
+statement ok
+ALTER TABLE regression_48233 RENAME TO reg_48233
+
+query IITI rowsort
+SELECT * FROM system.namespace WHERE name LIKE '%48233'
+----
+50  65  reg_48233  79
+
+statement error cannot convert a temporary table to a persistent table during renames
+ALTER TABLE reg_48233 RENAME TO public.reg_48233
+
+statement ok
+CREATE TABLE persistent_48233(a int)
+
+statement error schema cannot be modified: "pg_temp"
+ALTER TABLE reg_48233 RENAME TO pg_temp.pers_48233
```

Comment:
this should be `persistent_48233 RENAME TO pg_temp` right? (we're not doing anything with that table otherwise)